docs/USAGE-GUIDE.md

```diff
@@ -536,6 +568,53 @@ callback can get any ClientHello infromation from the connection and use
 The callback can return 0 to continue handshake in s2n or it can return negative
 value to make s2n terminate handshake early with fatal handshake failure alert.
 
+## Client Auth Related calls
```

Comment:
I think we agree that this client auth solution is too complex for *normal* users. Should we add some wording here to discourage use of this API until we can provide a simpler default? Alternatively, we can call out exactly what is going to be required of users to use mutual auth with s2n:
- complete x509 parsing in the application
- application knowledge of TLS code points for certificate types
- application dependency on libcrypto to give a libcrypto `RSA` struct back to s2n